File: LeetCode/Medium/3100-water-bottles-ii/3100-water-bottles-ii.py
class Solution:
    def maxBottlesDrunk(self, numBottles: int, numExchange: int) -> int:

        drunk_bottles: int = numBottles
        empty_bottles: int = numBottles

        while empty_bottles >= numExchange:
            drunk_bottles += 1
            empty_bottles -= numExchange - 1
            numExchange += 1
        
        return drunk_bottles

        

File: LeetCode/Medium/3100-water-bottles-ii/3100-water-bottles-ii.cpp
class Solution {
public:
    int maxBottlesDrunk(int numBottles, int numExchange) {

        int emptyBottles = numBottles;
        int drunkBottles = numBottles;

        while(emptyBottles >= numExchange){
            drunkBottles += 1;
            emptyBottles -= numExchange - 1;
            numExchange += 1;
        }

        return drunkBottles;
        
    }
};